From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Meyer <49727155+katexochen@users.noreply.github.com>
From: Sam Joan Roque-Worcel <sroque-worcel@gitlab.com>
Date: Mon, 22 Apr 2024 11:59:22 +0200
Subject: [PATCH] nixpkgs: use system C/C++ toolchains

Signed-off-by: Paul Meyer <49727155+katexochen@users.noreply.github.com>
Signed-off-by: Luke Granger-Brown <git@lukegb.com>
---
 bazel/dependency_imports.bzl | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/bazel/dependency_imports.bzl b/bazel/dependency_imports.bzl
index c508c60402..f79f2ba222 100644
--- a/bazel/dependency_imports.bzl
+++ b/bazel/dependency_imports.bzl
@@ -45,7 +45,11 @@ def envoy_dependency_imports(
         # - https://github.com/envoyproxy/envoy/issues/38951
         cargo_bazel_lockfile = "@envoy//:Cargo.Bazel.lock"):
     compatibility_proxy_repo()
-    rules_foreign_cc_dependencies()
+    rules_foreign_cc_dependencies(
+        register_default_tools=False,  # no prebuilt toolchains
+        register_built_tools=False,  # nor from source
+        register_preinstalled_tools=True,  # use host tools (default)
+    )
     go_rules_dependencies()
     go_register_toolchains(go_version)
     if go_version != "host":
